src/render/renderer.cpp

```diff
@@ -175,8 +175,23 @@ glm::vec4 Renderer::traceRayMIP(const Ray& ray, float sampleStep) const
 // Use the bisectionAccuracy function (to be implemented) to get a more precise isosurface location between two steps.
 glm::vec4 Renderer::traceRayISO(const Ray& ray, float sampleStep) const
 {
-    static constexpr glm::vec3 isoColor { 0.8f, 0.8f, 0.2f };
-    return glm::vec4(isoColor, 1.0f);
+    glm::vec3 sample_pos = ray.origin + ray.tmin * ray.direction;
+    const glm::vec3 increment = sampleStep * ray.direction;
+    glm::vec4 color(0.0f);
+
+    for (float t = ray.tmin; t <= ray.tmax; t += sampleStep, sample_pos += increment) {
+        auto voxel_value = this->m_pVolume->getVoxelInterpolate(sample_pos);
+        if (voxel_value > this->m_config.isoValue) {
+            color = this->getTFValue(voxel_value);
+            // color = glm::vec4 { 0.8f, 0.8f, 0.2f, 1.0f };

```

Comment:
Final vals returned aren't the isoColor

Reply:
I know, but I think this will anyway be overwritten in the final version, right?